packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -294,22 +299,49 @@ class VisualStudio {
         ...requirementArguments,
       ], encoding: encoding);
       if (whereResult.exitCode == 0) {
-        final List<Map<String, dynamic>> installations =
-            (json.decode(whereResult.stdout) as List<dynamic>).cast<Map<String, dynamic>>();
-        if (installations.isNotEmpty) {
+        final List<Map<String, dynamic>>? installations = _tryDecodeVswhereJson(whereResult.stdout);
+        if (installations != null && installations.isNotEmpty) {
           return VswhereDetails.fromJson(validateRequirements, installations[0]);
         }
       }
     } on ArgumentError {
       // Thrown if vswhere doesn't exist; ignore and return null below.
     } on ProcessException {
       // Ignored, return null below.
-    } on FormatException {
-      // may be thrown if invalid JSON is returned.
     }
     return null;
   }
 
+  List<Map<String, dynamic>>? _tryDecodeVswhereJson(String vswhereJson) {
+    late List<dynamic> result;
+    late FormatException error;
+    try {
+      // vswhere.exe is known to encode its output incorrectly, resulting in
+      // invalid JSON in the description property when interpreted as UTF-8.
+      // First, try to decode without any pre-processing.
+      // See: https://github.com/flutter/flutter/issues/106601
+      try {
+        result = json.decode(vswhereJson) as List<dynamic>;
+      } on FormatException catch (e) {
+        error = e;
+        vswhereJson = vswhereJson.replaceFirst(_vswhereDescriptionProperty, '');
```

Comment:
Since you have a "First, try to decode without any pre-processing." comment above I would add a "If that fails, remove the 'description' and try again.". I would also move the issue reference from above down to this new comment.

Reply:
This definitely improves clarity, thanks! Fixed!